extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -957,7 +961,8 @@ private List<SegmentIdWithShardSpec> generateSegmentIdsWithShardSpecs(
       final RowSignature signature,
       final ClusterBy clusterBy,
       final ClusterByPartitions partitionBoundaries,
-      final boolean mayHaveMultiValuedClusterByFields
+      final boolean mayHaveMultiValuedClusterByFields,
+      final Boolean isStageOutputEmpty
```

Comment:
I guess this is capital-B because `null` means unknown— in that case please annotate it with `@Nullable` and explain in the javadoc what `null` means. (It's rarely obvious to people without studying the code, and we'd like that to not be necessary whenever possible)

& do the same thing in the methods that this one calls out to.

Or, the other option is to define an enum, like `Capable`. Probably shouldn't use `Capable` since it's not really the right semantics, but it's a technique that aids in understanding.

Reply:
Thanks, added `@Nullable` annotations and docs to the relevant methods.